### PR TITLE
ACM-14164: Handle DHCP deployments

### PR DESCRIPTION
**Pull request by sakhoury**
This PR adds conditional guards before attempting to create NodeNetwork-related manifests.

### CODE REVIEW

File: internal/templates/assisted-installer/template.go [hunk 1]
```diff
@@ -141,7 +141,8 @@ spec:
   searchCollector:
     enabled: false`
 
-const NMStateConfig = `apiVersion: agent-install.openshift.io/v1beta1
+const NMStateConfig = `{{ if .SpecialVars.CurrentNode.NodeNetwork }}
```
Comment (carbonin) [inline internal/templates/assisted-installer/template.go:144]: Do we have some kind of unit test for this or no?

**Review by carbonin — APPROVED** [review]
Let's put an issue together to try to create some kind of basic unit tests on the templates directly. Maybe just rendering each template object with a crafted map of variables. Then validate the object parses into the type we want with the expected values.

File: internal/templates/assisted-installer/template.go (hunk 1 above)
Comment (sakhoury) [reply internal/templates/assisted-installer/template.go:144]: Created https://issues.redhat.com/browse/CNF-14484 for adding test coverage for default AI and IBI templates!

**Pull request merged**